guava/src/com/google/common/primitives/Bytes.java

```diff
@@ -152,6 +152,43 @@ private static int lastIndexOf(byte[] array, byte target, int start, int end) {
   }
 
   /**
+   * Returns the first position of the last occurrence of the specified {@code
+   * target} within {@code array}, or {@code -1} if there is no such occurrence.
+   *
+   * <p>More formally, returns the greatest index {@code i} such that
+   * {@code Arrays.copyOfRange(array, i, i + target.length)} contains exactly the same elements as
+   * {@code target}.
+   *
+   * @param array the array to search for the sequence {@code target}
+   * @param target the array to search for as a sub-sequence of {@code array}
+   */
+  public static int lastIndexOf(byte[] array, byte[] target) {
+    checkNotNull(array, "array");
+    checkNotNull(target, "target");
+    if (target.length == 0) {
+      return array.length - 1;
+    } else if (target.length > array.length) {
+      return -1;
+    }
+
+    int lastIndexOf = -1;
+    boolean differentValue;
+    for (int i = 0; i <= array.length - target.length; i++) {
```

Comment:
The way this search is implemented here, it is highly inefficient because it searches through every single match from the beginning of the array and keeps the last one. Instead it'd be wiser to do the `i` indexing in reverse like `Bytes.lastIndexOf(byte[], byte)` does it. Such approach also avoids the need for helper variables which you've used here.